### DISCUSSION TITLE
numberOfMonths withCaptionElement.

### DISCUSSION THREAD

**Discussion by jordan-jarolim** [gpbl/react-day-picker#1129] [Support]
Hi all, 

I ran into one issue I am not sure how to solve. It has been already discussed here: https://github.com/gpbl/react-day-picker/issues/623. The problem is in combination of `numberOfMonths` with `captionElement` in `DayPickerInput` component. 

When using `numberOfMonths > 1`, the `captionElement` (sent via `dayPickerProps`) is rendered for each of displayed months (DatePickers). However, I am using `YearMonthForm` from your doc as caption element and it does not make sense to have 2 YearMonthForms in one OverlayElement (even if it contains 2 DatePickers) as they both do the same action (navigate through OverlayElement's 2 DatePickers). So it would be nice to somehow address captionElement to specific DatePicker or add captionElement as overlay prop. 

Is there any other ways how to solve this use case?

Thanks

**Comment by gpbl** [comment]
Sorry could you explain your issue better? I’m not sure I understand. Some code examples?

**Comment by durasj** [comment]
Hi @gpbl, I think I know what @jordan-jarolim is trying to say. I am in a similar situation where I would like to display my custom `captionElement` only for the first calendar or handle the caption differently for each. I guess it would be enough if the DayPicker would also pass the currently rendered month index. Then I could od something like `captionElement={(props) => props.month === 0 ? <First> : <Others>}`.

**Comment by speir-wang** [comment]
@jordan-jarolim Hey buddy, have you solved this issue? I am having the same problem. 

Cheers

-------------------------

I found the answer, just use the `date` from `captionElement` prop so you will know how to target the calendar
